Add support for archive based articles (like Kolibri's HTML5 zip files)

dnohales opened the pull request:
https://phabricator.endlessm.com/T29258

dnohales committed:
Add support for archive based articles (like Kolibri's HTML5 zip files)

dylanmccall commented on dmodel/dm-article.h:
```diff
@@ -37,6 +37,12 @@ DM_AVAILABLE_IN_ALL
+dm_article_get_archive_member_content_stream (DmArticle *self,
+                                              const char *member_name,
+                                              GError **error);
+
```
This public interface makes me a little uneasy. Did you consider adding a separate DmArchive content type? Could inherit from either DmContent or DmArticle. We'd need our object metadata (as used in `dm_model_from_json_node`) to specify a different type for archives, but then we'd be able to add a separate DmArchive content type. DmArchive could inherit from either DmContent or DmArticle - whichever makes sense. I think that would make this a bit more natural to use from other places.

dnohales replied:
That would not actually follow the convention, since we use the same content type for PDFs and HTML besides they are substantially different kind of files, I kind of used the same logic for this...

dylanmccall replied:
Oooh, I see that now. Cool beans! The meat of this looks good to me :) Perhaps we can wiggle things around in the future.

dylanmccall commented:
I was going to suggest adding a test for this (along with new test data for archive articles), but I actually got carried away doing that myself since it's an opportunity to go through and fix some sdk-6 things :) I should have that to contribute here later today.

dylanmccall committed:
Add tests for get_archive_member_content_stream

To support this, I am adding another test content shard which contains a
application/x-kolibri-html5-zip file.

dylanmccall commented:
Okay, this looks good to me. I added a test in the most recent commit. Let me know if that covers everything you'd expect :)
